kernel/src/process_loading.rs

```diff
@@ -562,11 +607,14 @@ impl<C: Chip, D: ProcessStandardDebug> SequentialProcessLoaderMachine<'_, C, D>
         match ret {
             Ok(pb) => match self.checker.check(pb) {
                 Ok(()) => {}
-                Err(e) => {
-                    self.client.map(|client| {
+                Err(e) => match self.get_current_client() {
+                    Some(client) => {
                         client.process_loaded(Err(ProcessLoadError::CheckError(e)));
-                    });
-                }
+                    }
+                    None => {
+                        debug!("Error: Device is in an unknown operating mode.")
+                    }
+                },
```

Comment:
Restore this, should just be `                    self.get_current_client().map(|client| {`